components/encoder/encoder.go

```diff
@@ -39,12 +53,14 @@ var Subtype = resource.NewSubtype(
 
 // A Encoder turns a position into a signal.
 type Encoder interface {
-	// TicksCount returns number of ticks since last zeroing
-	TicksCount(ctx context.Context, extra map[string]interface{}) (float64, error)
+	// GetPosition returns number of ticks since last zeroing
+	GetPosition(ctx context.Context, positionType *pb.PositionType, extra map[string]interface{}) (float64, pb.PositionType, error)
```

Comment:
In general, I think we want to avoid directly passing automatically generated structs in what is essentially SDK interface code because it makes it difficult to support backward compatibility and accommodate API changes. The solution is to have a wrapper struct that can be generated from pb.PositionType. This conversion code would only be called by client/server code, making the codebase easier to update in the face of API changes. The wrapper struct also allows us to implement things on top of the position type, like validate that the value is in the proper domain for the type specified